tests/base/src/test/java/org/keycloak/tests/db/AbstractDBSchemaTest.java

```diff
@@ -0,0 +1,38 @@
+package org.keycloak.tests.db;
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.keycloak.admin.client.resource.RolesResource;
+import org.keycloak.representations.idm.RoleRepresentation;
+import org.keycloak.testframework.annotations.InjectClient;
+import org.keycloak.testframework.annotations.KeycloakIntegrationTest;
+import org.keycloak.testframework.config.Config;
+import org.keycloak.testframework.database.TestDatabase;
+import org.keycloak.testframework.realm.ManagedClient;
+import org.keycloak.testsuite.util.RoleBuilder;
+
+@KeycloakIntegrationTest
+public abstract class AbstractDBSchemaTest {
+
+    @InjectClient
+    ManagedClient managedClient;
+
+    protected static String dbType() {
+        String database = Config.getSelectedSupplier(TestDatabase.class);
+        return database == null ? "dev-mem" : database;
```

Comment:
Should not hard-code the default database here, as that could change. If this PR is merged after https://github.com/keycloak/keycloak/pull/41358, then it can be updated to:
```suggestion
        return Extensions.getInstance().findSupplierByType(TestDatabase.class).getAlias();
```